internal/api/keppel/accounts.go

```diff
@@ -1013,3 +1015,129 @@ func (a *API) handlePostAccountSublease(w http.ResponseWriter, r *http.Request)
+	authz := a.authenticateRequest(w, r, accountScopeFromRequest(r, keppel.CanChangeAccount))
+	if authz == nil {
+		return
+	}
+	account := a.findAccountFromRequest(w, r, authz)
+	if account == nil {
+		return
+	}
+
+	respondwith.JSON(w, http.StatusOK, map[string]any{"policies": json.RawMessage(account.TrivyPoliciesJSON)})
+}
+
+func (a *API) handlePutTrivyPolicies(w http.ResponseWriter, r *http.Request) {
+	httpapi.IdentifyEndpoint(r, "/keppel/v1/accounts/:account/trivy_policies")
+	authz := a.authenticateRequest(w, r, accountScopeFromRequest(r, keppel.CanChangeAccount))
+	if authz == nil {
+		return
+	}
+	account := a.findAccountFromRequest(w, r, authz)
+	if account == nil {
+		return
+	}
+
+	//decode existing policies
+	var dbPolicies []keppel.TrivyPolicy
+	err := json.Unmarshal([]byte(account.TrivyPoliciesJSON), &dbPolicies)
+	if respondwith.ErrorText(w, err) {
+		return
+	}
+
+	//decode request body
+	var req struct {
+		Policies []keppel.TrivyPolicy `json:"policies"`
+	}
+	decoder := json.NewDecoder(r.Body)
+	decoder.DisallowUnknownFields()
+	err = decoder.Decode(&req)
+	if err != nil {
+		http.Error(w, "request body is not valid JSON: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	//apply computed values and validate each input policy on its own
+	currentUserName := authz.UserIdentity.UserName()
+	var errs errext.ErrorSet
+	for idx, policy := range req.Policies {
+		path := fmt.Sprintf("policies[%d]", idx)
+		errs.Append(policy.Validate(path))
+
+		managingUserName := req.Policies[idx].ManagingUserName
+		switch managingUserName {
+		case "$REQUESTER":
+			req.Policies[idx].ManagingUserName = currentUserName
+		case "", currentUserName:
+			//acceptable
+		default:
+			if !slices.Contains(dbPolicies, policy) {
+				errs.Addf("cannot apply this new or updated policy that is managed by a different user: %s", policy)
+			}
+		}
+	}
+
+	//check that updated or deleted policies are either unmanaged or managed by
+	//the requester
+	for _, dbPolicy := range dbPolicies {
+		if slices.Contains(req.Policies, dbPolicy) {
+			continue
+		}
+		managingUserName := dbPolicy.ManagingUserName
+		if managingUserName != "" && managingUserName != currentUserName {
+			errs.Addf("cannot update or delete this existing policy that is managed by a different user: %s", dbPolicy)
+		}
+	}
+
+	//report validation errors
+	if !errs.IsEmpty() {
+		http.Error(w, errs.Join("\n"), http.StatusUnprocessableEntity)
+		return
+	}
+
+	//update policies in DB
+	jsonBuf, err := json.Marshal(req.Policies)
+	if respondwith.ErrorText(w, err) {
+		return
+	}
+	_, err = a.db.Exec(`UPDATE accounts SET trivy_policies_json = $1 WHERE name = $2`,
+		string(jsonBuf), account.Name)
+	if respondwith.ErrorText(w, err) {
+		return
+	}
+
+	//generate audit events
+	submitAudit := func(action cadf.Action, target audittools.TargetRenderer) {
+		if userInfo := authz.UserIdentity.UserInfo(); userInfo != nil {
+			a.auditor.Record(audittools.EventParameters{
+				Time:       time.Now(),
```

Comment:
Should we use the mocked time now here?

Reply:
This follows the existing implementation of audit events in other API handlers. Audit events need more than just time mocked out (e.g. each event also gets a v4 UUID for identification within Hermes). Therefore, we have a [normalization pass](https://github.com/sapcc/keppel/blob/3cb0cf024964539af63994e0481c4b4dcde376b7/internal/test/mock_auditor.go#L81) in the mock auditor that takes care of removing this real time value.